src/loaders/anndata-loaders/MatrixZarrLoader.js

```diff
@@ -0,0 +1,180 @@
+   */
+  async loadCellXGene() {
+    const { store, url } = this;
+    if (this.arr) {
+      return this.arr;
+    }
+    const {
+      options: { matrix },
+    } = this;
+    const res = await fetch(`${url}/${matrix.replace('.', '/')}/.zattrs`);
+    if (res.status !== 404) {
+      const zattrs = await res.json();
+      if (zattrs['encoding-type'] === 'csr_matrix') {
+        this.arr = this._loadCSRSparseCellXGene(matrix.replace('.', '/'), zattrs.shape);
+        return this.arr;
+      }
+      if (zattrs['encoding-type'] === 'csc_matrix') {
+        this.arr = this._loadCSCSparseCellXGene(matrix.replace('.', '/'), zattrs.shape);
+        return this.arr;
+      }
+    }
+    this.arr = openArray({ store, path: matrix.replace('.', '/'), mode: 'r' }).then(z => new Promise((resolve) => {
+      z.getRaw(null)
+        .then(resolve);
+    }));
+    return this.arr;
+  }
+
+  /**
+   * Class method for loading the genes list from AnnData.var.
+   * @returns {Promise} A promise for the zarr array contianing the gene names.
+   */
+  async loadGeneNames() {
+    if (this.geneNames) {
+      return this.geneNames;
+    }
+    const { genesFilter: genesFilterZarr } = this.options;
+    let genesFilter;
+    if (genesFilterZarr) {
+      const genesFilterArr = await openArray({
+        store: `${this.url}/${genesFilterZarr.replace('.', '/')}`,
+        mode: 'r',
+      });
+      const genesBufferCompressed = await genesFilterArr.store.getItem('0');
+      genesFilter = await genesFilterArr.compressor.decode(genesBufferCompressed);
+    }
+    const res = await fetch(`${this.url}/var/.zattrs`);
+    const { _index } = await res.json();
+    if (this.geneNames) {
+      return this.geneNames;
+    }
+    this.geneNames = openArray({
+      store: `${this.url}/var/${_index}`,
+      mode: 'r',
+    }).then(z => z.store
+      .getItem('0')
+      .then(buf => new Uint8Array(buf))
+      .then(cbytes => z.compressor.decode(cbytes))
+      .then((dbytes) => {
+        const text = this.decodeTextArray(dbytes)
+          .filter(i => !Number(i))
+          .filter(i => i.length >= 2)
+          .filter((_, j) => !genesFilter || genesFilter[j]);
+        return text;
+      }));
+    return this.geneNames;
+  }
+
+  load() {
+    return Promise
+      .all([this.loadCellNames(), this.loadGeneNames(), this.loadCellXGene()])
+      .then((d) => {
+        const [cellNames, geneNames, { data: cellXGeneMatrix }] = d;
+        const attrs = { rows: cellNames, cols: geneNames };
+        if (!this.min || !this.max) {
+          let max = -Infinity;
+          let min = Infinity;
+          for (let i = 0; i < cellXGeneMatrix.length; i += 1) {
+            const val = cellXGeneMatrix[i];
+            if (val > max) {
+              max = val;
+            } else if (val < min) {
+              min = val;
+            }
+          }
+          this.min = min;
+          this.max = max;
```

Comment:
```suggestion
          [this.min, this.max] = extent(cellXGeneMatrix);
```

with `import { extent } from 'd3-array';`

https://github.com/d3/d3-array/blob/master/src/extent.js looks very similar